ssl/t1_lib.c

```diff
@@ -2451,7 +2436,7 @@ static int ssl_scan_serverhello_tlsext(SSL *s, PACKET *pkt, int *al)
         else if (type == TLSEXT_TYPE_application_layer_protocol_negotiation) {
             unsigned len;
             /* We must have requested it. */
-            if (s->alpn_client_proto_list == NULL) {
+            if (!s->s3->alpn_sent || s->alpn_client_proto_list == NULL) {
```

Comment:
Well, can we find out for certain and act accordingly? :)

Sorry for being picky but it'll make it easier to beat sense into the SSL structure in the future. Consider my pickiness an appreciation of the pull request :)